src/Vision/Image.php

```diff
@@ -0,0 +1,331 @@
+     * $image = new Image($imageResource, [
+     *     'faces',      // Corresponds to `FACE_DETECTION`
+     *     'landmarks',  // Corresponds to `LANDMARK_DETECTION`
+     *     'logos',      // Corresponds to `LOGO_DETECTION`
+     *     'labels',     // Corresponds to `LABEL_DETECTION`
+     *     'text',       // Corresponds to `TEXT_DETECTION`
+     *     'safeSearch', // Corresponds to `SAFE_SEARCH_DETECTION`
+     *     'properties'  // Corresponds to `IMAGE_PROPERTIES`
+     * ]);
+     * ```
+     *
+     * @param  resource|string|Object $image An image to configure with the given
+     *         settings. This parameter will accept a resource, a string of
+     *         bytes, or an instance of {@see Google\Cloud\Storage\Object}.
+     * @param  array $features A list of cloud vision
+     *         [features](https://cloud.google.com/vision/reference/rest/v1/images/annotate#type)
+     *         to apply to the image. gcloud-php provides a set of abbreviated
+     *         names which can be used in the interest of brevity in place of
+     *         the names offered by the cloud vision service. These names are
+     *         `faces`, `landmarks`, `logos`, `labels`, `text`, `safeSearch`
+     *         and `properties`.
+     * @param  array $options {
+     *     Configuration Options
+     *
+     *     @type array $maxResults A list of features and the maximum number of
+     *           results to return. Keys should correspond to feature names
+     *           given in the `$features` array, and values should be of type
+     *           int. In all cases where `$maxResults` does not contain a value
+     *           for a feature, all results will be returned. In cases where
+     *           a `$maxResults` value is specified, the cloud vision service
+     *           will return results up to the `$maxResults` value, or the full
+     *           results, whichever is fewer.
+     *     @type array $imageContext See
+     *           [ImageContext](https://cloud.google.com/vision/reference/rest/v1/images/annotate#imagecontext)
+     *           for full usage details.
+     * }
+     * @throws InvalidArgumentException
+     */
+    public function __construct($image, array $features, array $options = [])
+    {
+        $this->options = $options + [
+            'imageContext' => [],
+            'maxResults' => []
+        ];
+
+        $this->features = $this->normalizeFeatures($features);
+
+        if ($image instanceof Object) {
+            $identity = $image->identity();
+            $uri = sprintf('gs://%s/%s', $identity['bucket'], $identity['object']);
+
+            $this->type = self::TYPE_STORAGE;
+            $this->image = $uri;
+        } elseif (is_string($image)) {
+            $this->type = self::TYPE_STRING;
+            $this->image = $image;
+        } elseif (is_resource($image)) {
+            $this->type = self::TYPE_BYTES;
+            $this->image = new Stream($image);
```

Comment:
Could we please use https://github.com/guzzle/psr7/blob/master/src/functions.php#L78 here instead? This way we won't have a hard requirement on guzzle if the transport ever needs to change.